crates/context/src/lib.rs

```diff
@@ -346,20 +346,28 @@ impl ContextManager {
 
     pub async fn join_context(
         &self,
-        identity_secret: PrivateKey,
         invitation_payload: ContextInvitationPayload,
     ) -> EyreResult<Option<(ContextId, PublicKey)>> {
         let (context_id, invitee_id, protocol, network_id, contract_id) =
             invitation_payload.parts()?;
 
-        if identity_secret.public_key() != invitee_id {
-            bail!("identity mismatch")
-        }
+        let placeholder_context_id = ContextId::from([0; 32]);
 
-        let handle = self.store.handle();
+        let stored_identity = self
+            .get_identity_value(placeholder_context_id, invitee_id)?
+            .ok_or_eyre(eyre::eyre!(
+                "Missing identity for public key: {}",
+                invitee_id
+            ))?;
 
-        let identity_key = ContextIdentityKey::new(context_id, invitee_id);
+        let main_private_key_bytes = stored_identity
```

Comment:
```suggestion
        let private_key = stored_identity
```

or `identity_secret`